staging/src/k8s.io/apimachinery/pkg/runtime/extension.go

```diff
@@ -18,10 +18,68 @@ package runtime
 
 import (
 	"bytes"
-	"encoding/json"
 	"errors"
+	"fmt"
+	"reflect"
+
+	cbor "k8s.io/apimachinery/pkg/runtime/serializer/cbor/direct"
+	"k8s.io/apimachinery/pkg/util/json"
+	"sigs.k8s.io/structured-merge-diff/v4/value"
```

Comment:
are we ok picking up dependencies on smd here? does this expose us to any of the json-iterator issues that repo has (https://github.com/kubernetes-sigs/structured-merge-diff/issues/202)

Reply:
Hmm, I read https://github.com/kubernetes/kubernetes/blob/010409ad0db507a94d8fd9a0528f27a50e525a47/vendor/sigs.k8s.io/structured-merge-diff/v4/value/reflectcache.go#L184-L185 and used this in favor of `runtime.DefaultUnstructuredConverter` to take advantage of self-converting types.

Re: Risk exposure, isn't this path already taken on objects received from clients? @jpbetz

Reply:
It's fine to call `DefaultUnstructuredConverter`, but it does depends on SMD to do:

https://github.com/kubernetes/kubernetes/blob/742b2f70b9e5c1ec03682ad25dc76fdcc7109310/staging/src/k8s.io/apimachinery/pkg/runtime/converter.go#L346-L349

Reply:
Ah, I remember now, I went directly to SMD for this instead of via `DefaultUnstructuredConverter` because RawExtension may contain something other than a map. Its ToUnstructured signature doesn't support it, and it will actually panic if you try:

https://github.com/kubernetes/kubernetes/compare/master...benluddy:defaultunstructuredconverter-non-mapstringany